batchglm/train/numpy/base_glm/estimator.py

```diff
@@ -86,8 +94,95 @@ def train_sequence(self, training_strategy, **kwargs):
                         "overrding %s from training strategy with value %s with new value %s\n"
                         % (x, str(d[x]), str(kwargs[x]))
                     )
+            if "dispersion_smoothing" in d:
+                if d["dispersion_smoothing"] == "sctransform":
+                    if FDataGrid is None or NWSmoother is None:
+                        logger.error("Missing optional dependency scikit-fda.")
             self.train(**d, **kwargs)
             logger.debug("Training sequence #%d complete", idx + 1)
+            if "dispersion_smoothing" in d:
+                if d["dispersion_smoothing"] == "sctransform":
+                    self.perform_vst()
+                elif d["dispersion_smoothing"] == "edger":
+                    raise NotImplementedError()
+                elif d["dispersion_smoothing"] == "deseq2":
+                    raise NotImplementedError()
+
+    def perform_vst(
+        self,
+        theta_regularization: str = "od_factor",
+        use_geometric_mean: bool = True,
+        gmean_eps: float = 1.0,
+        bw_adjust: float = 1.5,
+    ):
+        logger.info("Performing Dispersion smoothing with flavour sctransform...")
+        # compute geometric log mean counts
+        if use_geometric_mean:
+            genes_log_gmean = np.log10(geometric_mean(self.model_container.x, axis=0, eps=gmean_eps))
+        else:
+            genes_log_gmean = np.log10(np.mean(self.model_container.x, axis=0))
+        if isinstance(genes_log_gmean, dask.array.core.Array):
+            genes_log_gmean = genes_log_gmean.compute()
+
+        # specify which kind of regularization is performed
+        scale_param = np.exp(self.model_container.theta_scale[0])  # TODO check if this is always correct
```

Comment:
why an exponentiation here? is this meant to be link-function specific?

Reply:
I'm not sure how to resolve this (thus the TODO). Indeed, I think this depends on the noise model. I will have to look at that again.